src/normalizers/regex.js

```diff
@@ -58,11 +58,11 @@ export function namedDecoder(name, decoder) {
 }
 
 // Generic pluggable encoder
-export function regexMatchingEncoderMaker(name, regex, charMap) {
-    const fn = function encoder(str) {
+export function regexMatchingEncoderMaker(name, regex, charMap, cb) {
+    const fn = function encoder(str, flags = {}) {
         return str.replaceAll(regex, (match, ...capture) => {
             const charToReplace = capture.reduce((p,c) => p ?? c);
-            return match.replace(charToReplace, charMap[charToReplace]);
+            return match.replace(charToReplace, cb? cb(name, charMap, charToReplace, flags) : charMap[charToReplace]);
```

Comment:
I thought we'd do `charMap[charToReplace] ?? (cb && cb(name, charToReplace, flags))` so the function kicks in only when there's no simple match -- in that case we don't need to pass charMap since by definition it doesn't have any match.

passing an unused charMap seems weird an unnecessarily assuming what the callback does.

Reply:
But if `charMap[charTo]' returns an object like `{ja: 'ホッペー', ko: '홉 따는 사람'}` then the callback won't be called.